src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.System/Main.cs

```diff
@@ -117,8 +218,29 @@ private List<Result> Commands()
                     IcoPath = $"Images\\sleep.{IconTheme}.png",
                     Action = c =>
                     {
-                        NativeMethods.SetSuspendState(false, true, true);
-                        return true;
+                        if (this._context.CurrentPluginMetadata.ConfirmSys)
+                        {
+                            MessageBoxResult messageBoxResult = MessageBox.Show(
+                                "You are about to put this computer to sleep, are you sure?",
+                                "Please Confirm",
+                                MessageBoxButton.YesNo,
+                                MessageBoxImage.Warning);
+
+                            if (messageBoxResult == MessageBoxResult.Yes)
+                            {
+                                NativeMethods.SetSuspendState(false, true, true);
+                                return true;
+                            }
+                            else
+                            {
+                                return false;
+                            }
+                        }
+                        else
+                        {
+                            NativeMethods.SetSuspendState(false, true, true);
+                            return true;
+                        }
```

Comment:
Confirmation dialog logic looks repetitive. So let's move it into `bool ExecuteCommand(string confirmationMessage, Func<bool> command)`.

Reply:
I had some trouble doing this with Func<bool>, but I was able to accomplish what I think you were wanting with Action.  If this is not acceptable, I would need some assistance with making this with Func<bool>.